### ISSUE TITLE
False positives from gradle build-dependencies

### ISSUE THREAD

**Issue by bkimminich** [dependency-check/DependencyCheck#519]
Gradle plugins - which are only used during the build process - should not get analyzed by default. Currently they are analyzed which causes false positives. Example:

```
    apply plugin: 'findbugs'`

    findbugs {
        toolVersion = '3.0.1'
        sourceSets = [sourceSets.main, sourceSets.test]
        effort = 'default'
        reportLevel = 'high'
    }
```

Running `gradle dependencyCheck` will now report vulnerabilities on `com.apple:AppleJavaExtensions:1.4` which is a dependency of the FindBugs plugin:

```
findbugs - The FindBugs libraries to be used for this project.
\--- com.google.code.findbugs:findbugs:3.0.1
     +--- net.jcip:jcip-annotations:1.0
     +--- com.google.code.findbugs:jsr305:2.0.1
     +--- com.google.code.findbugs:bcel-findbugs:6.0
     +--- com.google.code.findbugs:jFormatString:2.0.1
     +--- dom4j:dom4j:1.6.1
     |    \--- xml-apis:xml-apis:1.0.b2
     +--- org.ow2.asm:asm-debug-all:5.0.2
     +--- org.ow2.asm:asm-commons:5.0.2
     |    \--- org.ow2.asm:asm-tree:5.0.2
     |         \--- org.ow2.asm:asm:5.0.2
     +--- commons-lang:commons-lang:2.6
     +--- com.apple:AppleJavaExtensions:1.4
     \--- jaxen:jaxen:1.1.6
```

It would be great to either never scan gradle plugins or introduce a new config parameter e.g. `skipBuildDependencies` to be able to turn that behavior off.

**Comment by riy** [comment]
I'm voting for the new parameter `skipBuildDependencies`, because it does make sense to scan build dependencies every now and then. Thanks to NPM's open nature malware etc. could creep in there as well, having access to the local filesystem.

**Comment by bkimminich** [comment]
Just realized that the Gradle plugin has its own repo and submitted [a PR to add `skipBuildGroups`](https://github.com/jeremylong/dependency-check-gradle/pull/9) there. In the same fashion like test dependencies are skipped this one just skips everything with `classpath` in the beginning, which is - as far as I know Gradle - only allowed `buildscript { dependencies { -->here<-- } }`.

**Comment by jeremylong** [comment]
A [PR (#8)](https://github.com/jeremylong/dependency-check-gradle/pull/8) was already merged to resolve this issue. This was included in the 1.4.0 release and allows one to configure either a `skipConfiguration` or `scanConfiguration` (mutually exclusive atm). 

In addition to this we are considering adding a default scanConfiguration of `runtime`.

Lastly, I just pushed an update to 1.4.1-SNAPSHOT that displays the `scope` that a dependency was identified in.

@riy - completely agree that even the build plugins should be scanned from time to time. I've been considering an update to both the Maven and Gradle plugins to support this.

As PR #8 was included in 1.4.0 I am going to consider this issue resolved.
